### PR TITLE
Get rid of numpy.distutils

### LINKED ISSUE #2148: Some tests still want numpy.distutils
Since `numpy` 1.26.0 removed `numpy.distutils` on Python 3.12, we are still seeing a few tests failing in Fedora Rawhide even with `pythran` 0.14.0, which includes https://github.com/serge-sans-paille/pythran/pull/2049.

```
=================================== FAILURES ===================================
___________________ TestDistutils.test_setup_bdist_install3 ____________________
[…]
E           subprocess.CalledProcessError: Command '['/usr/bin/python3', 'setup.py', 'bdist', '--dist-dir=bdist']' ret
urned non-zero exit status 1.

/usr/lib64/python3.12/subprocess.py:413: CalledProcessError
----------------------------- Captured stderr call -----------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/pythran-0.14.0/pythran/tests/test_distutils_numpy/setup.py", line 1, in <module>
    from numpy.distutils.core import setup
ModuleNotFoundError: No module named 'numpy.distutils'
_______________________ TestDistutils.test_setup_build3 ________________________
[…]
E           subprocess.CalledProcessError: Command '['/usr/bin/python3', 'setup.py', 'build']' returned non-zero exit status 1.

/usr/lib64/python3.12/subprocess.py:413: CalledProcessError
----------------------------- Captured stderr call -----------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/pythran-0.14.0/pythran/tests/test_distutils_numpy/setup.py", line 1, in <module>
    from numpy.distutils.core import setup
ModuleNotFoundError: No module named 'numpy.distutils'
___________________ TestDistutils.test_setup_sdist_install3 ____________________
[…]
E           subprocess.CalledProcessError: Command '['/usr/bin/python3', 'setup.py', 'sdist', '--dist-dir=sdist3']' returned non-zero exit status 1.

/usr/lib64/python3.12/subprocess.py:413: CalledProcessError
----------------------------- Captured stderr call -----------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/pythran-0.14.0/pythran/tests/test_distutils_numpy/setup.py", line 1, in <module>
    from numpy.distutils.core import setup
ModuleNotFoundError: No module named 'numpy.distutils'
```

Please consider skipping these tests when `numpy.distutils` is not available.

### PR DESCRIPTION
**Pull request by serge-sans-paille**
Since numpy 1.26.0, numpy.distutils is being removed in python 3.12, let's try to have a solution that work for 3.12 *and* previous version without numpy.distutils.

Fix #2148

### CODE REVIEW

File: docs/DEVGUIDE.rst [hunk 1]
```diff
@@ -68,8 +68,12 @@ Validation
 ----------
 
 ``pythran`` uses the ``unittest`` module and the `pytest
-<http://pytest.org/latest/>`_ package to manage test cases. The whole
-validation suite is run through the command::
+<http://pytest.org/latest/>`_ package to manage test cases.
+
+All requirementsa r elisted in ``pythran/tests/requirements.txt``.
```
Comment (hroncok) [inline docs/DEVGUIDE.rst:73]:
```suggestion
All requirements are listed in ``pythran/tests/requirements.txt``.
```

**Review by rgommers — APPROVED** [review]
This worked fine for SciPy, and the changes LGTM.

**Comment by serge-sans-paille** [comment]
Thanks @hroncok, @musicinmybrain  and @rgommers  for extra-testing this patch.

**Comment by reneeotten** [comment]
@serge-sans-paille would you please consider releasing a new version that contains these changes?

**Comment by serge-sans-paille** [comment]
@reneeotten : https://github.com/serge-sans-paille/pythran/pull/2167

**Pull request merged**